php/php.editor/nbproject/org-netbeans-modules-php-editor.sig

```diff
@@ -1648,11 +1658,11 @@ meth public int start_state()
 meth public java.lang.String getFileName()
 meth public java_cup.runtime.Symbol do_action(int,java_cup.runtime.lr_parser,java.util.Stack,int) throws java.lang.Exception
 meth public org.netbeans.modules.php.editor.parser.ParserErrorHandler getErrorHandler()
-meth public org.netbeans.modules.php.editor.parser.astnodes.VariableBase createDispatch(boolean,org.netbeans.modules.php.editor.parser.astnodes.VariableBase,org.netbeans.modules.php.editor.parser.astnodes.Expression,int,int,java.util.List<org.netbeans.modules.php.editor.parser.astnodes.Expression>,int,java.util.List,java.util.List)
-meth public org.netbeans.modules.php.editor.parser.astnodes.VariableBase createDispatch(org.netbeans.modules.php.editor.parser.astnodes.VariableBase,org.netbeans.modules.php.editor.parser.astnodes.VariableBase)
-meth public org.netbeans.modules.php.editor.parser.astnodes.VariableBase createDispatch(org.netbeans.modules.php.editor.parser.astnodes.VariableBase,org.netbeans.modules.php.editor.parser.astnodes.VariableBase,java.util.List)
-meth public org.netbeans.modules.php.editor.parser.astnodes.VariableBase createDispatch(org.netbeans.modules.php.editor.parser.astnodes.VariableBase,org.openide.util.Pair<org.netbeans.modules.php.editor.parser.astnodes.Expression,java.lang.Boolean>,java.util.List)
-meth public org.openide.util.Pair<org.netbeans.modules.php.editor.parser.astnodes.Expression,java.lang.Boolean> createDispatchProperty(boolean,org.netbeans.modules.php.editor.parser.astnodes.Expression,int,int,java.util.List<org.netbeans.modules.php.editor.parser.astnodes.Expression>,int,java.util.List)
```

Comment:
Looks like an incompatible API change. Technically it should be associated with increasing the version of the module to `/2` to signal incompatibilities. E.g. modules that used to link to previous version of `VariableBase.createDispatch` are no longer going to link with the new version. Probably harmless. I don't expect many modules linking against `php.editor` being out there.

CC @junichi11

Reply:
Maybe, these methods don't have to be public... (I can be wrong because I am not the original author.)
Is the following OK? (I haven't run the unit tests yet but these are not used other places, I suppose.)
<details>
<summary>patch</summary>

```patch
diff --git a/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Parser.java b/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Parser.java
index b6fb4a7f2d..10b37ed610 100644
--- a/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Parser.java
+++ b/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Parser.java
@@ -19,7 +19,7 @@
 
 //----------------------------------------------------
 // The following code was generated by CUP v0.11a beta 20060608
-// Fri Sep 04 08:35:13 JST 2020
+// Sat Oct 31 20:05:48 JST 2020
 //----------------------------------------------------
 
 package org.netbeans.modules.php.editor.parser;
@@ -30,7 +30,7 @@ import org.netbeans.modules.php.editor.parser.astnodes.*;
 import org.openide.util.Pair;
 
 /** CUP v0.11a beta 20060608 generated parser.
-  * @version Fri Sep 04 08:35:13 JST 2020
+  * @version Sat Oct 31 20:05:48 JST 2020
   */
 @org.netbeans.api.annotations.common.SuppressWarnings({"EI_EXPOSE_REP", "MS_PKGPROTECT", "BC_BAD_CAST_TO_CONCRETE_COLLECTION"})
 public class ASTPHP5Parser extends java_cup.runtime.lr_parser {
@@ -2009,7 +2009,15 @@ new EncodedActionTable15().getTableData()
         return anonymousClassCounter;
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, Pair<Expression, Access> pair, List dimensions) {
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, Pair<Expression, Boolean> pair, List dimensions) {
+        Pair<Expression, Access> access = Pair.of(pair.first(), pair.second() ? Access.STATIC : Access.NON_STATIC);
+        return createDispatch(dispatcher, dimensions, access);
+    }
+
+    VariableBase createDispatch(VariableBase dispatcher, List dimensions, Pair<Expression, Access> pair) {
         VariableBase dispatch = null;
         Expression property = pair.first();
         Access access = pair.second();
@@ -2061,15 +2069,38 @@ new EncodedActionTable15().getTableData()
         return dispatch;
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions, Access access) {
-        return createDispatch(dispatcher, Pair.of(property, access), null);
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions) {
+        return createDispatch(dispatcher, null, Pair.of(property, Access.NON_STATIC));
+    }
+
+    VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions, Access access) {
+        return createDispatch(dispatcher, null, Pair.of(property, access));
+    }
+
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property) {
+        return createDispatch(dispatcher, property, null, Access.NON_STATIC);
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, Access access) {
+    VariableBase createDispatch(VariableBase dispatcher, VariableBase property, Access access) {
         return createDispatch(dispatcher, property, null, access);
     }
 
-    public VariableBase createDispatch(Access access, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(boolean isStatic, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+            List<Expression> paramsList, int paramsListright, List propertyList, List aa) {
+        Access access = isStatic ? Access.STATIC : Access.NON_STATIC;
+        return createDispatch(access, var, memberProperty, memberPropertyleft, memberPropertyright, paramsList, paramsListright, propertyList, aa);
+    }
+
+    VariableBase createDispatch(Access access, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
             List<Expression> paramsList, int paramsListright, List propertyList, List aa) {
         Expression firstVarProperty = null;
         if (paramsList == null) {
@@ -2090,14 +2121,24 @@ new EncodedActionTable15().getTableData()
         Iterator listIt = list.iterator();
         while (listIt.hasNext()) {
             Pair<Expression, Access> property = (Pair<Expression, Access>) listIt.next();
-            dispatch = createDispatch(dispatcher, property, arrayDimensiones);
+            dispatch = createDispatch(dispatcher, arrayDimensiones, property);
             dispatcher = dispatch;
             arrayDimensiones = new LinkedList();
         }
         return dispatch;
     }
 
-    public Pair<Expression, Access> createDispatchProperty(Access access, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+    /**
+     * @deprecated
+     */
+    public Pair<Expression, Boolean> createDispatchProperty(boolean isStatic, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+            List<Expression> paramsList, int paramsListright, List aa) {
+        Access access = isStatic ? Access.STATIC : Access.NON_STATIC;
+        Pair<Expression, Access> pair = createDispatchProperty(access, memberProperty, memberPropertyleft, memberPropertyright, paramsList, paramsListright, aa);
+        return Pair.of(pair.first(), isStatic);
+    }
+
+    Pair<Expression, Access> createDispatchProperty(Access access, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
             List<Expression> paramsList, int paramsListright, List aa) {
         Expression result = null;
         if (paramsList == null) {
diff --git a/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Symbols.java b/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Symbols.java
index 8b0053039a..bb12526053 100644
--- a/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Symbols.java
+++ b/php/php.editor/src/org/netbeans/modules/php/editor/parser/ASTPHP5Symbols.java
@@ -19,7 +19,7 @@
 
 //----------------------------------------------------
 // The following code was generated by CUP v0.11a beta 20060608
-// Fri Sep 04 08:35:13 JST 2020
+// Sat Oct 31 20:05:48 JST 2020
 //----------------------------------------------------
 
 package org.netbeans.modules.php.editor.parser;
diff --git a/php/php.editor/tools/ASTPHP5Parser.cup b/php/php.editor/tools/ASTPHP5Parser.cup
index eb77e0848c..cf14ed89b0 100644
--- a/php/php.editor/tools/ASTPHP5Parser.cup
+++ b/php/php.editor/tools/ASTPHP5Parser.cup
@@ -67,7 +67,15 @@ parser code {:
         return anonymousClassCounter;
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, Pair<Expression, Access> pair, List dimensions) {
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, Pair<Expression, Boolean> pair, List dimensions) {
+        Pair<Expression, Access> access = Pair.of(pair.first(), pair.second() ? Access.STATIC : Access.NON_STATIC);
+        return createDispatch(dispatcher, dimensions, access);
+    }
+
+    VariableBase createDispatch(VariableBase dispatcher, List dimensions, Pair<Expression, Access> pair) {
         VariableBase dispatch = null;
         Expression property = pair.first();
         Access access = pair.second();
@@ -119,15 +127,38 @@ parser code {:
         return dispatch;
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions, Access access) {
-        return createDispatch(dispatcher, Pair.of(property, access), null);
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions) {
+        return createDispatch(dispatcher, null, Pair.of(property, Access.NON_STATIC));
+    }
+
+    VariableBase createDispatch(VariableBase dispatcher, VariableBase property, List dimensions, Access access) {
+        return createDispatch(dispatcher, null, Pair.of(property, access));
+    }
+
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property) {
+        return createDispatch(dispatcher, property, null, Access.NON_STATIC);
     }
 
-    public VariableBase createDispatch(VariableBase dispatcher, VariableBase property, Access access) {
+    VariableBase createDispatch(VariableBase dispatcher, VariableBase property, Access access) {
         return createDispatch(dispatcher, property, null, access);
     }
 
-    public VariableBase createDispatch(Access access, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+    /**
+     * @deprecated
+     */
+    public VariableBase createDispatch(boolean isStatic, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+            List<Expression> paramsList, int paramsListright, List propertyList, List aa) {
+        Access access = isStatic ? Access.STATIC : Access.NON_STATIC;
+        return createDispatch(access, var, memberProperty, memberPropertyleft, memberPropertyright, paramsList, paramsListright, propertyList, aa);
+    }
+
+    VariableBase createDispatch(Access access, VariableBase var, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
             List<Expression> paramsList, int paramsListright, List propertyList, List aa) {
         Expression firstVarProperty = null;
         if (paramsList == null) {
@@ -148,14 +179,24 @@ parser code {:
         Iterator listIt = list.iterator();
         while (listIt.hasNext()) {
             Pair<Expression, Access> property = (Pair<Expression, Access>) listIt.next();
-            dispatch = createDispatch(dispatcher, property, arrayDimensiones);
+            dispatch = createDispatch(dispatcher, arrayDimensiones, property);
             dispatcher = dispatch;
             arrayDimensiones = new LinkedList();
         }
         return dispatch;
     }
 
-    public Pair<Expression, Access> createDispatchProperty(Access access, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+    /**
+     * @deprecated
+     */
+    public Pair<Expression, Boolean> createDispatchProperty(boolean isStatic, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
+            List<Expression> paramsList, int paramsListright, List aa) {
+        Access access = isStatic ? Access.STATIC : Access.NON_STATIC;
+        Pair<Expression, Access> pair = createDispatchProperty(access, memberProperty, memberPropertyleft, memberPropertyright, paramsList, paramsListright, aa);
+        return Pair.of(pair.first(), isStatic);
+    }
+
+    Pair<Expression, Access> createDispatchProperty(Access access, Expression memberProperty, int memberPropertyleft, int memberPropertyright,
             List<Expression> paramsList, int paramsListright, List aa) {
         Expression result = null;
         if (paramsList == null) {
```
</details>
cc: @tmysik

Reply:
@junichi11 

AFAIR there is a PHP Editor API module. However, I think that also the PHP Editor module had to be a friend of some other PHP module(s) as it was not easily possible to "transfer" the needed functionality into the API module.

However, because it is an API friend (if I am not wrong), increasing the major version (and updating the relevant PHP modules) should not be a problem, I guess.

cc @jtulach

Reply:
@tmysik Thank you for your help!

I looked for "ASTPHP5Parser".
```
$ cd netbeans
$ find . -name '*.java' | xargs grep "ASTPHP5Parser"
```

It seems to be used in the php.editor module only.

`createDispatch*` methods are used in the `ASTPHP5Parser` only.

Reply:
Well, it was in the API from some reason (repo history could help here) so I think that any change would mean increasing the major version.

cc @jtulach

Reply:
> Probably harmless. I don't expect many modules linking against `php.editor` being out there.

Up to you guys. As I said. An incompatible change in `php` cluster probably is not as dangerous as incompatible change in `ide` or `java` cluster.

Reply:
OK, thank you guys :) I'll increase the major version.